tools/cross/format.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python3
+
+import logging
+import os
+import re
+import subprocess
+from argparse import ArgumentParser, Namespace
+from typing import List, Optional
+
+CLANG_FORMAT = os.environ.get("CLANG_FORMAT", "clang-format")
+
+
+def parse_args() -> Namespace:
+    parser = ArgumentParser()
+    parser.add_argument(
+        "--check",
+        help="only check for files requiring formatting; don't actually format them",
+        action="store_true",
+        default=False,
+    )
+    subparsers = parser.add_subparsers(dest="subcommand")
+    git_parser = subparsers.add_parser(
+        "git", help="Apply format to changes tracked by git"
+    )
+    git_parser.add_argument(
+        "--source",
+        help="consider files modified in the specified commit-ish; if not specified, defaults to all changes in the working directory",
+        type=str,
+        required=False,
+        default=None,
+    )
+    git_parser.add_argument(
+        "--target",
+        help="consider files modified since the specified commit-ish; defaults to HEAD",
+        type=str,
+        required=False,
+        default="HEAD",
+    )
+    git_parser.add_argument(
+        "--staged",
+        help="consider files with staged modifications only",
+        action="store_true",
+        default=False,
+    )
+    options = parser.parse_args()
+    if options.subcommand == "git":
+        if options.staged:
+            if options.source is not None or options.target != "HEAD":
+                logging.error(
+                    "--staged cannot be used with --source or --target; use --staged with --source=HEAD"
+                )
+                exit(1)
+    return options
+
+
+def check_clang_format() -> bool:
+    try:
+        # Run clang-format with --version to check its version
+        output = subprocess.check_output([CLANG_FORMAT, "--version"]).decode("utf-8")
```

Comment:
Pass an encoding argument rather than decoding the output:
```suggestion
        output = subprocess.check_output([CLANG_FORMAT, "--version"], encoding="utf-8")
```